lib/db_tests.js

```diff
@@ -179,4 +205,106 @@ describe('Database API', function() {
     });
   });
 
+  /* Advanced listing - restrictions
+   {"eq": { "a": 1 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"ne": { "a": 1 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------";
+   {"lt": { "a": 2 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"le": { "a": 2 }}       data.count === 2 ? "SUCCESS!!!!!" : "--------FAILURE--------";
+   {"gt": { "a": 1 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"ge": { "a": 1 }}       data.count === 2 ? "SUCCESS!!!!!" : "--------FAILURE--------";
+   {"in": { "a": [1] }}     data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"sort": { "a": -1 }}    data.list[0].fields.a === 2 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"skip": 0, "limit": 1}  data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   */
+
+  it('advanced listing - Equals 1', function (done) {
+    dbListPromisified(type, {"eq": { "a": 1 }})
+      .then(function(data) {
+        expect(data.count).to.equal(1);
```

Comment:
IMO the expected results should be defined as constants at the beginning of the `describe` block, instead of summarising them in a comment block. The way it's written now could easily lead to confusion and human error. What I am saying is something like:
```
const AdvancedListingRestrictions = {
  equalTo = 1;
}

[...]
expect(data.count).to.equal(AdvancedListingResults.equalTo);

Reply:
this is actually good idea and I'll add it to the code 👍